pkg/trace/writer/sender.go

```diff
@@ -351,6 +351,15 @@ func newPayload(headers map[string]string) *payload {
 	return p
 }
 
+func (p *payload) clone() *payload {
+	clone := ppool.Get().(*payload)
```

Comment:
Should we use `newPayload` here? It'd get rid of some duplication.
```
headers := make(map[string]string, len(p.headers))
for k, v := range p.headers {
    headers[k] = v
}
clone := newPayload(headers)
clone.body.ReadFrom(bytes.NewBuffer(p.body.Bytes()))
```
And let's clone the headers just to avoid any potential programming errors in the future. I think we can take this cost since this is not really a documented nor production-ready feature.